### PR TITLE
docs(dependency-bots): mention PEP 723 + some improvements

### PR DESCRIPTION
**Pull request by mkniewallner**
## Summary

Renovate recently gained support for updating dependencies defined using PEP 723 (https://github.com/renovatebot/renovate/pull/31266). Since uv supports this format, I thought it could be nice to mention support for it in the integrations documentation as well. I took the occasion to make the page a bit more structured as well.

## Test Plan

Ran Renovate on https://github.com/mkniewallner/renovate-pep723, which created https://github.com/mkniewallner/renovate-pep723/pull/2 that updates a dependency defined using PEP 723. But I'll re-run some tests again once the changes are released on Renovate cloud GitHub app just in case.

### CODE REVIEW

**Review by zanieb — APPROVED** [review]

**Comment by zanieb** [comment]
Thanks!

**Comment by mkniewallner** [comment]
Renovate version has been updated in the cloud app today. Just in case, tested the feature on the app and it works as expected: https://github.com/mkniewallner/renovate-pep723/pull/4.

File: docs/guides/integration/dependency-bots.md [hunk 1]
```diff
@@ -16,7 +29,7 @@ dependencies) by enabling the
 [`lockFileMaintenance`](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance)
 option:
 
-```json5 title="renovate.json5"
+```jsx title="renovate.json5"
```
Comment (konstin) [inline docs/guides/integration/dependency-bots.md:32]: Why the change to jsx here?

Comment (mkniewallner) [reply docs/guides/integration/dependency-bots.md:32]: - `json5`:

Screenshot From 2024-10-09 21-10-28

- `jsx`:

Screenshot From 2024-10-09 21-10-36

Comment (konstin) [reply docs/guides/integration/dependency-bots.md:32]: Interesting catch!
